### PR TITLE
Add docs regarding Azure Sphere, currently not linked to the docs site.

### PR DESCRIPTION
**Pull request by spygi**
# PR for issue #1456

## What is being addressed

Initial investigation on Azure Sphere and whether that could be useful for the starterkit.

### CODE REVIEW

File: docs/tools/azure-sphere.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# What is Azure Sphere
+
+[Azure Sphere](https://azure.microsoft.com/en-us/services/azure-sphere/) delivers device security by
+combining hardware, operating system software, and cloud services that has been purpose-built for
```
Comment (roel4ez) [inline docs/tools/azure-sphere.md:4]:
```suggestion
combining hardware, operating system software, and cloud services that have been purpose-built for
```

Comment (spygi) [reply docs/tools/azure-sphere.md]: paragraph was rewritten.

**Comment by roel4ez** [comment]
Proposal: move to Spikes folder

File: docs/spikes/azure-sphere.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+combining hardware, OS-level software, and cloud services to secure IoT applications. Together these
+three components ensure [The Seven Properties of Highly Secure
+Devices](https://www.microsoft.com/en-us/research/uploads/prod/2020/11/Seven-Properties-of-Highly-Secured-Devices-2nd-Edition-R1.pdf).
+
+[Pricing](https://azure.microsoft.com/en-us/pricing/details/azure-sphere/): one time purchase of
+hardware only (no recurring fees)
+
+## Advantages for the starter kit
+
+- Automated management of the underlying OS and the concentrator logic.
```
Comment (bastbu) [inline docs/spikes/azure-sphere.md:13]: What do you mean by concentrator logic? Maybe a brief explanation would be helpful.

File: docs/spikes/azure-sphere.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+## Disadvantages
+
+- Requires specific hardware with [limited choices so
+  far/potentially more expensive](https://azure.microsoft.com/en-us/services/azure-sphere/#ecosystem) (some also out of stock)
+  - There is a [LoRaWAN concentrator built from
+    Miromico](https://www.avnet.com/wps/portal/silica/solutions/technologies/wireless-connectivity/lora-gateways/)
+    that only works with the Packet Forwarder for now. Miromico are aware of this limitation but we
+    are not sure when they plan to support LBS.
+- Introduces another layer that needs to be considered.
```
Comment (bastbu) [inline docs/spikes/azure-sphere.md:31]: Is there a potential restriction with respect to which Azure Sphere features we can use if the concentrators cannot connect to the internet? So far, they only need to connect to the LNS, maybe it's a restriction worth mentioning.

File: docs/spikes/azure-sphere.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+## Advantages for the starter kit
+
+- Automated management of the underlying OS and the concentrator logic.
+- Although some features are already supported e.g. password-less authentication, we could benefit
+  from built-in, e.g:
+  - hardware security
+  - error reporting (i.e. A software error, such as a buffer overrun induced by an attacker probing
+security, is reported to cloud-based failure analysis system)
+  - renewable security (revoking compromised assets for known vulnerabilities)
```
Comment (bastbu) [inline docs/spikes/azure-sphere.md:19]: I don't understand this point, I think it would be worth expanding a bit on what "renewable security" is and what "revoking compromised assets" means.

File: docs/spikes/azure-sphere.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  hardware could be justified or even be cheaper in the long term.
+
+## Disadvantages
+
+- Requires specific hardware with [limited choices so
+  far/potentially more expensive](https://azure.microsoft.com/en-us/services/azure-sphere/#ecosystem) (some also out of stock)
+  - There is a [LoRaWAN concentrator built from
+    Miromico](https://www.avnet.com/wps/portal/silica/solutions/technologies/wireless-connectivity/lora-gateways/)
+    that only works with the Packet Forwarder for now. Miromico are aware of this limitation but we
+    are not sure when they plan to support LBS.
```
Comment (bastbu) [inline docs/spikes/azure-sphere.md:30]: [nit] I would emphasize the fact that it currently only works with the Packet Forwarder, maybe it's worth adding as a separate bullet point.

File: docs/spikes/azure-sphere.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+- Get the hardware (either a new device or "guardian module" that is "attached" to existing hardware)
+- Install SDK/CLI + [VS
+  extension](https://marketplace.visualstudio.com/items?itemName=AzureSphereTeam.AzureSphereSDKforVisualStudio2019)
+  and connect the device to the PC
+- [Create a
+  "tenant"](https://docs.microsoft.com/en-us/azure-sphere/deployment/manage-tenants?tabs=cliv2beta)
+- [Claim the
+  device](https://docs.microsoft.com/en-us/azure-sphere/install/claim-device?tabs=cliv2beta):
+  one-time operation that can not be undone
+- [Configure networking](https://docs.microsoft.com/en-us/azure-sphere/install/configure-wifi)
```
Comment (bastbu) [inline docs/spikes/azure-sphere.md:46]: [nit] Instead of duplicating the steps from the Azure Sphere quickstart, which might get outdated potentially, I think it's good enough to just link to the official quickstart.

If you think it makes sense to capture what the steps were at the time of writing, it's also fine to resolve the comment.

File: docs/spikes/azure-sphere.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+- [Subscribe to
+  notifications/updates](https://docs.microsoft.com/en-us/azure-sphere/install/get-notifications)
+
+## Integration with other Azure services
+
+Can be used [together with IoT Hub or IoT
```
Comment (bastbu) [inline docs/spikes/azure-sphere.md:52]: Do I understand it correctly that integration with IoT Hub mostly offers us simplification/functionality with respect to authentication of concentrators? Or are there any other IoT Hub functionalities that we could use when integrating with Azure Sphere (besides all the benefits we get that are not related to IoT Hub)?

**Review by bastbu — APPROVED** [review]
Thanks for the investigation, LGTM!

File: docs/spikes/azure-sphere.md (hunk 3 above)
Comment (spygi) [reply docs/spikes/azure-sphere.md:31]: Good point, added it as a disadvantage.

File: docs/spikes/azure-sphere.md (hunk 2 above)
Comment (spygi) [reply docs/spikes/azure-sphere.md:13]: I meant the LBS images. Reworded it to be more explicit.

File: docs/spikes/azure-sphere.md (hunk 4 above)
Comment (spygi) [reply docs/spikes/azure-sphere.md]: Good point, simplified it to "automated security updates" instead

File: docs/spikes/azure-sphere.md (hunk 6 above)
Comment (spygi) [reply docs/spikes/azure-sphere.md:68]: Fair point but because we had some notes that are specific to us, I think the short recap makes sense

File: docs/spikes/azure-sphere.md (hunk 7 above)
Comment (spygi) [reply docs/spikes/azure-sphere.md]: My understanding is that we would still need IoTHub for certain features (the twins, message persistence, downstream responses). Azure Sphere will help with maintenance and ensuring secure updates.
